src/wire/Socket.ts

```diff
@@ -249,10 +249,14 @@ export class InteractiveSocket extends EventEmitter {
             return;
         }
 
-        this.state = SocketState.Closing;
-        this.socket.close(1000, 'Closed normally.');
-        this.queue.forEach(packet => packet.cancel());
-        this.queue.clear();
+        if (this.state !== SocketState.Idle) {
+            this.state = SocketState.Closing;
+            this.socket.close(1000, 'Closed normally.');
+            this.queue.forEach(packet => packet.cancel());
+            this.queue.clear();
+        } else {
+            this.state = SocketState.Closing;
```

Comment:
Actually, well. It would make sense to transition back to Idle. Functionally closed and idle would be identical. Ignore my previous comment, then 😛